src/test/java/org/jenkinsci/plugins/gitclient/GitUsernamePasswordBindingTest.java

```diff
@@ -0,0 +1,181 @@
+
+        //GitUsernamePasswordBinding instance
+        gitCredBind = new GitUsernamePasswordBinding(credentials.getId());
+        assertEquals("Type mis-match", StandardUsernamePasswordCredentials.class, gitCredBind.type());
+
+
+        if (gitToolName == null) {
+            gitToolName = gitToolInstance.getGitExe();
+        } else {
+            gitToolInstance = getToolInstance(gitToolName);
+        }
+        //Git
+        Git git = Git.with(TaskListener.NULL, new EnvVars()).in(gitRootRepo).using(getToolInstance(gitToolName).getGitExe());
+
+        //Git Tool
+        Jenkins.get().getDescriptorByType(GitTool.DescriptorImpl.class).getDefaultInstallers().clear();
+        Jenkins.get().getDescriptorByType(GitTool.DescriptorImpl.class).setInstallations(gitToolInstance);
+    }
+
+    public GitTool getToolInstance(String name) {
+        GitTool tool = null;
+        switch (name) {
+            case "git":
+                tool = GitTool.getDefaultInstallation();
+                break;
+            case "jgit":
+                tool = JGitTool.getDefaultInstallation();
+                break;
+            case "jgitapache":
+                tool = JGitApacheTool.getDefaultInstallation();
+        }
+        return tool;
+    }
+
+    @Test
+    public void test_GenerateGitScript_write() throws IOException, InterruptedException {
+        GitUsernamePasswordBinding.GenerateGitScript tempGenScript = new GitUsernamePasswordBinding.GenerateGitScript(this.username, this.password, credentials.getId());
+        assertEquals("Type mis-match", StandardUsernamePasswordCredentials.class, tempGenScript.type());
+        FilePath tempScriptFile = tempGenScript.write(credentials, rootFilePath);
+        assertEquals("Read and Execute permissions to be set:" + tempScriptFile.mode(), 320, tempScriptFile.mode());
+        if (!isWindows()) {
+            assertEquals("File extension not sh", "sh", FilenameUtils.getExtension(tempScriptFile.getName()));
+        } else {
+            assertEquals("File extension not bat", "bat", FilenameUtils.getExtension(tempScriptFile.getName()));
+        }
+        assertTrue("Username not found", tempScriptFile.readToString().contains(this.username));
+        assertTrue("Password not found", tempScriptFile.readToString().contains(this.password));
+    }
+
+    @Test
+    public void test_FreeStyleProject() throws Exception {
+        FreeStyleProject prj = r.createFreeStyleProject();
+        prj.getBuildWrappersList().add(new SecretBuildWrapper(Collections.<MultiBinding<?>>
+                singletonList(new GitUsernamePasswordBinding(credentialID))));
+        if (isWindows()) {
+            prj.getBuildersList().add(new BatchFile("@echo off\necho %Git_Username%:%Git_Password% > auth.txt"));
+        } else {
+            prj.getBuildersList().add(new Shell("set +x\necho $Git_Username:$Git_Password > auth.txt"));
```

Comment:
I would expect this test to fail because I thought that shell variable names were case sensitive.  Since the shell variable name is `GIT_USERNAME` it seems like `Git_Username` should resolve to the empty string.